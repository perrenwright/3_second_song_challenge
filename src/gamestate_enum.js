const GAME_STATE = {
    BEFORE: 'before',
    IN_PROGRESS: 'in_progress',
    ENDED: 'ended',
}
export {GAME_STATE};