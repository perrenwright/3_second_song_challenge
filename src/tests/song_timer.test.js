import React from "react";
import Song_timer from "../components/song_timer.js"
import {GAME_STATE} from '../gamestate_enum.js';
import { render, fireEvent, act, getByTestId, cleanup } from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';

afterEach(cleanup)

test('Song_timer button sets length', () => {

  let setgameTime = jest.fn();

  const { getByTestId } = render(<Song_timer gameTime={3000} setgameTime={setgameTime} gameState={null}/>);

    act(() => {
      fireEvent.click(getByTestId("button1"));
    });

    expect(setgameTime).toHaveBeenCalledWith(3000);
});

test('Song_timer button displays options', () => {

  let setgameTime = jest.fn();

  const { getByText, getByTestId } = render(<Song_timer gameTime={3000} setgameTime={setgameTime} gameState={null}/>);

    act(() => {
      fireEvent.click(getByTestId("button1"));
    });
    const linkElement = getByText("3 Seconds");
    expect(linkElement).toBeInTheDocument();
});

test('Song_timer button displays options', () => {

  let setgameTime = jest.fn();

  const { getByText, getByTestId } = render(<Song_timer gameTime={3000} setgameTime={setgameTime} gameState={null}/>);

    act(() => {
      fireEvent.click(getByTestId("button1"));
    });
    const linkElement = getByText("5 Seconds");
    expect(linkElement).toBeInTheDocument();
});

test('Song_timer button displays options', () => {

  let setgameTime = jest.fn();

  const { getByText, getByTestId } = render(<Song_timer gameTime={3000} setgameTime={setgameTime} gameState={null}/>);

    act(() => {
      fireEvent.click(getByTestId("button1"));
    });
    const linkElement = getByText("10 Seconds");
    expect(linkElement).toBeInTheDocument();
});