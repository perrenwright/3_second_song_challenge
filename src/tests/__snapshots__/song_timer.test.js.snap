// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song timer component Matches the snapshot 1`] = `<div />`;
